modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/ClusterProcessor.java

```diff
@@ -607,6 +610,33 @@ private Boolean findLastFlag(Collection<Serializable> vals) {
                 U.error(log, "Failed to register MBean for cluster: ", e);
             }
         }
+
+        MetricRegistry reg = ctx.metric().registry(CLUSTER_METRICS);
+
+        reg.register("TopologyVersion",
```

Comment:
Can we use this metric and all below inside `ClusterMetricsMXBeanImpl` to keep single method to calculate metric value.

Reply:
We can use this metrics for the `ClusterMetricsMXBeanImpl` implementation. But not for the `ClusterLocalNodeMetricsMXBeanImpl`.
Moreover metrics registration should be placed between starting `GridMetricManager` and beans registration. For example, in the `IgniteKernal#registerMetrics()`